src/librustc_metadata/cstore.rs

```diff
@@ -82,13 +82,19 @@ pub struct CrateMetadata {
     pub dep_kind: Lock<DepKind>,
     pub source: CrateSource,
 
-    pub proc_macros: Option<Vec<(ast::Name, Lrc<SyntaxExtension>)>>,
+    pub proc_macros: Option<Vec<FullProcMacro>>,
 
     /// Whether or not this crate should be consider a private dependency
     /// for purposes of the 'exported_private_dependencies' lint
     pub private_dep: bool
 }
 
+pub struct FullProcMacro {
```

Comment:
Maybe call this `ProcMacroDef`?

Reply:
There's already a struct called `ProcMacroDef` in `src/libsyntax_ext/proc_macro_harness.rs`. I didn't want to have two different structs with the same name.